File: 프로그래머스/lv2/12980. 점프와 순간 이동/점프와 순간 이동.py
def solution(n):
    answer = 0
    # 뒤로 갈수 없고 앞으로만 가기 때문에 dp로 풀지 않아도 된다
    # 슈트를 입은 사람이 N에 도착한 것을 반대로 한다고 생각하면 된다.. 
    while n > 0:
        if n % 2 == 0:
            n = n // 2
        else:
            answer += 1
            n -= 1 
    
    return answer

File: 프로그래머스/lv2/12980. 점프와 순간 이동/점프와 순간 이동.java
import java.util.*;

public class Solution {
    public int solution(int n) {
        return move(n);
    }
    
    public int move(int N){
        int result = 0;
        while(N>0){
            if(N%2 != 0){
                N-= 1;
                result += 1;
            }else {
                N /= 2;
            }
        }
        
        return result;
    }
}
